docs/architecture/engine.md

```diff
@@ -0,0 +1,197 @@
+# Storage Engine Overview
+
+M3DB is a time series database that was primarily designed to be horizontally scalable and handle a large volume of monitoring time series data.
+
```

Comment:
Could you add a table of contents with links to the various sub-sections here.

Reply:
The UI generates a table of contents automatically

Reply:
dope